test/Common/AutoFixture/Attributes/RepeatingPatternBitAutoDataAttribute.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Reflection;
+
+namespace Bit.Test.Common.AutoFixture.Attributes;
+
+public class RepeatingPatternBitAutoDataAttribute : BitAutoDataAttribute
```

Comment:
💭 
I really like the idea of what we're trying to solve here. It's annoying to type out and easy to miss some of those inline data sets. 

I think it would have been easier to have this in a separate PR, since it's introducing a new testing tool/attribute. 



Xunit has a few different solutions with their idea of sharing test data `ClassData` `MemberData` and I know there are a few libraries that similarly do what you have proposed.


For this specifically, I think combining xUnit's idea of `ClassData` with `BitAutoData` is an alternative to consider here. 

Something like (this is rough code could be cleaned up)

```
  public class ClassBitAutoDataAttribute : BitAutoDataAttribute
    {
        public ClassBitAutoDataAttribute(Type @class)
        {
            Class = @class;
        }

        public Type Class { get; private set; }

        public override IEnumerable<object[]> GetData(MethodInfo testMethod)
        {
            IEnumerable<object[]> data = Activator.CreateInstance(Class) as IEnumerable<object[]>;
            if (data == null)
            {
                throw new ArgumentException(
                    string.Format(
                        CultureInfo.CurrentCulture,
                        "{0} must implement IEnumerable<object[]> to be used as ClassData for the test method named '{1}' on {2}",
                        Class.FullName,
                        testMethod.Name,
                        testMethod.DeclaringType.FullName
                    )
                );
            }

            foreach (var classData in data)
            {
                var bitData = base.GetData(testMethod).First();
                for (var i = 0; i < classData.Length; i++)
                {
                    bitData[i] = classData[i];
                }

                yield return bitData;
            }
        }
    }
```


Then we could have one class to share data across tests

```
   public class SendAsyncTestData : TheoryData<bool, bool, bool>
    {
        public SendAsyncTestData()
        {
            Add(false, false, false);
            Add(true, false, false);
            Add(false, true, false);
            Add(true, true, false);
            Add(false, false, true);
            Add(true, false, true);
            Add(false, true, true);
        }
    };
```

Usage in an individual test.

```
    [Theory]
    [ClassBitAutoDataAttribute(typeof(SendAsyncTestData))]
    public async Task SendAsync_UserIdSet_SendPayloadToUserAsync(bool haveIdentifier, bool haveDeviceId,
        bool haveOrganizationId, SutProvider<PushController> sutProvider, Guid installationId, Guid userId,
        Guid identifier, Guid deviceId)
    {
```

This would give us more flexibility and allow for generating (looping through combinations) the test data within the test data class if required.

Reply:
Separated to #5167

I don't like the `SendAsyncTestData` suggestion, since it obscures the data from the test, so it's harder to know what we are actually testing. And i know we will have `SendAsyncTestData2Booleans`, `SendAsyncTestData3Booleans` and other more complex data's that will be hard to describe.

Reply:
Agreed on the naming and repeat of class test data, but that is just a general concern of using that XUnit feature.


If obscurity is a concern, I don't see how either option is an improvement. Tracking down what data was used in the test that is intermittently failing is a problem we face with using BitAutoData in the current state.